crates/node/src/main.rs

```diff
@@ -76,14 +76,27 @@ async fn main() -> Result<()> {
             sqlx::migrate!().run(&pool).await.map_err(|e| e.into())
         }
         Command::Peer { peer, op } => match op {
-            PeerCommand::Whitelist { whitelist } => {
-                todo!("implement peer whitelisting");
+            PeerCommand::Whitelist { db_url } => {
+                let db = storage::Database::new(&db_url).await?;
+                let key = entity::PublicKey::try_from(peer.as_str())?;
+                db.acl_whitelist(&key).await
             }
-            PeerCommand::Deny { deny } => {
-                todo!("implement peer denying");
+            PeerCommand::Deny { db_url } => {
+                let db = storage::Database::new(&db_url).await?;
+                let key = entity::PublicKey::try_from(peer.as_str())?;
+                db.acl_deny(&key).await
             }
         },
         Command::Run { config } => run(Arc::new(config)).await,
+        Command::Show { op } => match op {
+            ShowCommand::PublicKey { key_file } => {
+                let bs = std::fs::read(key_file)?;
+                let key = SecretKey::parse(bs.as_slice().try_into()?)?;
+                let public_key = PublicKey::from_secret_key(&key);
+                println!("{}", hex::encode(public_key.serialize()));
```

Comment:
Use log::trace! instead of println!

Reply:
Actually, isn't `println!` totally fine in this case as it's purely meant to be "write to stdout" for end user, no matter what the logging configuration is?

Reply:
Ah ok, I didn't really investigate the reason, I thought it was one forgets.